### ISSUE TITLE
Google Fit steps being overwritten

### ISSUE THREAD

**Issue by zym88** [praveendath92/fitbit-googlefit#28]
Great app!

I use both my phone and my Fitbit separately throughout the day depending on the circumstances. I noticed that when I run this sync program, it overwrites all of the steps data on Google Fit. 

The other apps that I have used that sync steps with Google Fit seem to be smarter and only overwrite for the time period the steps occurred and not the whole day. Is it possible to facilitate this?

Thank you!
Ken

PS, the instructions for the Google API need to be updated. They have changed their requirements.

**Comment by praveendath92** [comment]
@zym88 Thanks for the feedback.

The tool itself doesn't "overwrite" any existing data synced by other apps to GoogleFit. However, it writes some steps value for every few seconds (and sometimes those are zeros from Fitbit) and I suppose GoogleFit is not properly handling when two different apps sync different data for the same time frame. 

In any case, I will check if I can add an option to skip timeframes where Fitbit reports zero steps. Hopefully, that will fix it. 

Last but not least, could you please point out what changed specifically? Thanks again 😄

**Comment by zym88** [comment]
Yes, I'm sure that skipping timeframes where Fitibit reports zero will do the trick. Thanks!

I had to go through a mandatory web site registration with the Google API interface. None of that was mentioned in the how-to. Just worth a heads-up. Also, Google won't allow the word "google" in the app name and Fitibit won't allow the name "fitibit" in the app name. Worth a mention since the errors were not clear right away and it took some trial and error for them to accept. Just my experience. YMMV.

Keep up the great work!

**Comment by praveendath92** [comment]
@zym88 I will try to fix this next week. Thanks again for bringing these 2 issues to my notice :+1:
